nav2_robot/include/nav2_robot/ros_robot.hpp

```diff
@@ -28,6 +28,7 @@ namespace nav2_robot
 class RosRobot : public Robot
 {
 public:
+  using Footprint = std::vector<geometry_msgs::msg::Point>;
```

Comment:
How about using a struct (with named fields, of course) so that there is no chance to use the wrong point. That is, the user of the Footprint doesn't have to know that item 0 is the frontLeft, for example?

Reply:
A footprint is an arbitrary vector or points, so each point doesn't mean anything in particular in the general case.

It just donned on me now, that probably 0,0 is the baselink, so all the points should be relative to that. It should match the URDF and all the other transforms on the robot. 

And then the X axis is forward by default.

I'm not sure if the points need to be ordered or not. 

This was meant as just a quick and dirty implementation so I could start migrating DWB to use the robot class, with a real footprint added in the near future.